tests/Rules/PHPUnit/DataProviderDataRuleTest.php

```diff
@@ -0,0 +1,202 @@
+<?php declare(strict_types=1);
+
+namespace PHPStan\Rules\PHPUnit;
+
+use PhpParser\Node;
+use PHPStan\Testing\CompositeRule;
+use PHPStan\Rules\Methods\CallMethodsRule;
+use PHPStan\Rules\Rule;
+use PHPStan\Testing\RuleTestCase;
+use PHPStan\Type\FileTypeMapper;
+
+/**
+ * @extends RuleTestCase<CompositeRule>
+ */
+class DataProviderDataRuleTest extends RuleTestCase
+{
+
+	protected function getRule(): Rule
+	{
+		$reflectionProvider = $this->createReflectionProvider();
+
+		/** @var list<Rule<Node>> $rules */
+		$rules = [
+			new DataProviderDataRule(
+				new TestMethodsHelper(
+					self::getContainer()->getByType(FileTypeMapper::class),
+					true
+				),
+				new DataProviderHelper(
+					$reflectionProvider,
+					self::getContainer()->getByType(FileTypeMapper::class),
+					self::getContainer()->getService('defaultAnalysisParser'),
+					true
+				),
+
+			),
+			self::getContainer()->getByType(CallMethodsRule::class) /** @phpstan-ignore phpstanApi.classConstant */
+		];
+
+		return new CompositeRule($rules);
+	}
+
+	public function testRule(): void
+	{
+		if (PHP_VERSION_ID < 80000) {
+			self::markTestSkipped();
```

Comment:
Why skip for PHP < 8.0?

Reply:
separated the named argument tests into a new file